### PR TITLE
Proposal for `gen` and `collapse`

### PR DESCRIPTION
**Pull request by pdeffebach**
This is a very naive implementation of @bkamins 's plan for `collapse`, etc. described [here](https://github.com/JuliaData/DataFrames.jl/issues/2206#issuecomment-619390620). 

Please keep comments focused on #2206. This is just to demonstrate how I imagine this should work.

### CODE REVIEW

**Comment by bkamins** [comment]
OK - I treat it as a reference to a discussion. This PR should not be merged and will be closed when #2206 is resolved.

**Comment by pdeffebach** [comment]
I will update this PR as consensus develops. When we all agree we can implement it for real.

**Comment by bkamins** [comment]
OK, but as @nalimilan commented in #2206 I do not expect that we will allow to define so many functions in DataFrames.jl. We will rather keep a minimal list of functions that cover all required functionality. The functions you propose are good for an extension package to DataFrames.jl so that users can opt-in to use them.

File: src/gen_proposal.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+function gen(df::AbstractDataFrame, args...; kwargs...)
+	transform(df, args...; kwargs...)
```
Comment (bkamins) [inline src/gen_proposal.jl:2]: first small note (just for the future discussions) - this is not a correct implementation as `args` could overwrite all source columns of `df`.

File: src/gen_proposal.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+end
+
+function genby(df::AbstractDataFrame, id, args...; kwargs...)
+	by(df, id, :, args...; kwargs...)
+end
+
+function keep(df::AbstractDataFrame, args...; kwargs...)
+		t = gensym()
```
Comment (bkamins) [inline src/gen_proposal.jl:10]: a second small note `gensym()` cannot be used here as it does not guarantee we will not have column name conflicts. But again - as above this is a minor issue than can be fixed.

**Comment by bkamins** [comment]
I am closing this as 0.21 was released. If you decide to put these definitions in some accompanying package ping me and I can help with the review. Thank you for working on this!

**Pull request closed without merging**